### DISCUSSION TITLE
$set('property', value) cannot set multiple properties

### DISCUSSION THREAD

**Discussion by parmonov98** [livewire/livewire#5011] [3. Feature ideas]
I'd like to set multiple properties value from js. 
$setAll() or $set(array variable) would be awesome. 

if I use $set multiple times it causes component multiple rendering which is fine by small and light components but when it comes to bigger ones and with calculating tasks  it 's a problem.

Thanks!

**Comment by joshhanley** [comment]
@parmonov98 set takes a third parameter, which is a boolean, and if set to true, can defer the update. So you could defer all of your sets, except the last one, which will then trigger the component to re-render. https://laravel-livewire.com/docs/2.x/alpine-js#interacting-with-livewire-from-alpine

```js
// Defer setting a Livewire property to a specific value
$wire.set('property', value, true)
```

Hope this helps!

**Reply by parmonov98** [reply]
> @parmonov98 set takes a third parameter, which is a boolean, and if set to true, can defer the update. So you could defer all of your sets, except the last one, which will then trigger the component to re-render. https://laravel-livewire.com/docs/2.x/alpine-js#interacting-with-livewire-from-alpine
> 
> ```js
> // Defer setting a Livewire property to a specific value
> $wire.set('property', value, true)
> ```
> 
> Hope this helps!

Thanks. Anyway, it's needed. Because sometimes we want to do  complex calculation and send the result at once.

**Reply by joshhanley** [reply]
@parmonov98 yep, hopefully it's a good workaround for now though. All PR's are welcome! 🙂
